src/fidesops/task/graph_task.py

```diff
@@ -410,16 +410,17 @@ def filter_data_categories(
             )
         )
 
-        if not target_fields:
+        if not target_field_paths:
             continue
 
-        for row in results:
-            filtered_access_results[node_address].append(
-                {
-                    field: result
-                    for field, result in row.items()
-                    if field in {target.string_path for target in target_fields}
-                }
-            )
+        # Normalize nested data into a flat dataframe
+        df = pd.json_normalize(results, sep=".")
+        # Only keep dataframe columns that match the target field paths
+        df = df[[field_path.string_path for field_path in target_field_paths]]
+        # Turn the filtered results back into a list of dictionaries
+        filtered_flattened_results = df.to_dict(orient="records")
+        for row in filtered_flattened_results:
+            # For each row, unflatten the dictionary
+            filtered_access_results[node_address].append(unflatten_dict(row))
```

Comment:
Using pandas `json_normalize` which we also use when we flatten data for CSV upload - wanted to get more use out of this tool since we include it as a dependency!

Unfortunately, I couldn't find a method that does the opposite of `json_normalize` and rebuilds the data in the nested format we want to return to the user, so I added an `unflatten_dict` util.